src/MapViewController.swift

```diff
@@ -125,165 +255,391 @@ open class MapViewController: UIViewController, LocationManagerDelegate {
+   
+   - parameter identifier: The marker to set.
+   - pameter polyline: The polyline to represent the marker.
+   - returns: A boolean if the marker set succeeded.
+   */
   open func markerSetPolyline(_ identifier: TGMapMarkerId, polyline: TGGeoPolyline) -> Bool {
     return tgViewController.markerSetPolyline(identifier, polyline: polyline)
   }
-  
+
+  /**
+   Sets the marker to represent a particular polygon.
+   
+   - parameter identifier: The marker to set.
+   - pameter polygon: The polygon to represent the marker.
+   - returns: A boolean if the marker set succeeded.
+   */
   open func markerSetPolygon(_ identifier: TGMapMarkerId, polygon: TGGeoPolygon) -> Bool {
     return tgViewController.markerSetPolygon(identifier, polygon: polygon)
   }
-  
+
+  /**
+   Sets the marker if it should be visible or not.
+   
+   - parameter identifier: The marker to alter visibility on.
+   - parameter visible: True means visible. False mean not visible.
+   - returns: A boolean indiciation whether or not the update succeeded.
+  */
   open func markerSetVisible(_ identifier: TGMapMarkerId, visible: Bool) -> Bool {
     return tgViewController.markerSetVisible(identifier, visible: visible)
   }
-  
+
+  /**
+   Sets the marker bitmap image.
+   
+   - parameter identifier: The marker to set the image for.
+   - parameter image: A UIImage to represent the marker.
+   - returns: A boolean indiciation whether or not the update succeeded.
+   */
   open func markerSetImage(_ identifier: TGMapMarkerId, image: UIImage) -> Bool {
     return tgViewController.markerSetImage(identifier, image: image)
   }
-  
+
+  /**
+   Removes an individual marker from the map.
+   
+   - parameter marker: The marker to remove.
+   - returns: A boolean indicating if the removal succeeded.
+  */
   open func markerRemove(_ marker: TGMapMarkerId) -> Bool {
     return tgViewController.markerRemove(marker)
   }
-  
+
+  /** 
+   Loads a scene file synchronously on the main thread. Use the async methods instead of these in production apps.
+   
+   - parameter path: The path to the scene file to load.
+   - throws: A MZError `apiKeyNotSet` error if an API Key has not been sent on the MapzenManager class.
+ */
   open func loadSceneFile(_ path: String) throws {
     try loadSceneFile(path, sceneUpdates: [TGSceneUpdate]())
   }
-  
+
+  /**
+   Loads a scene file synchronously on the main thread. Use the async methods instead of these in production apps.
+   
+   - parameter path: The path to the scene file to load.
+   - parameter sceneUpdates: The scene updates to make while loading the scene file.
+   - throws: A MZError `apiKeyNotSet` error if an API Key has not been sent on the MapzenManager class.
+  */
   open func loadSceneFile(_ path: String, sceneUpdates: [TGSceneUpdate]) throws {
     try tgViewController.loadSceneFile(path, sceneUpdates: updatesWithApiKeyUpdate(sceneUpdates))
   }
 
+  /**
+   Loads the scene file asynchronously. Recommended for production apps.
```

Comment:
We might want to recommend that the user wait to make modifications to the map until only after they receive the closure callback